### PR TITLE
15 Handling Events

### PR DESCRIPTION
**Pull request by magdalena-u**
Chapter 15: Handling Events
15_1 Balloon
15_2 Mouse trail
15_3 Tabs

### CODE REVIEW

File: 15_1_baloon.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const pump = event => {
+    if (event.keyCode === 38) {
+        maximize();
+    }
+
+    if (event.keyCode === 40) {
+        minimize();
+    }
+
+    if (baloon.style.fontSize === 250 + 'px') {
```
Comment (AdamusPiotr) [inline 15_1_baloon.js]: use templateString

File: 15_1_baloon.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        boom();
+    }
+};
+
+const maximize = () => {
+    size += 10;
+    level += 10;
+    baloon.style.fontSize = size + 'px';
```
Comment (AdamusPiotr) [inline 15_1_baloon.js]: use template string with `${size}px`

Comment (AdamusPiotr) [reply 15_1_baloon.js]: And change it everywhere u can

File: 15_2_trail.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+};
+const remove = () => {
+    array.forEach(item => {
+        const index = array.indexOf(item);
+        array.splice(index, 1);
+    });
+};
+
+const loop = () => {
+    requestAnimationFrame(loop);
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: I'm not sure but requestAnimationFrame shouldn't be under the draw ?

File: 15_2_trail.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    draw();
+};
+
+class Element {
+    constructor(x, y) {
+        this.x = x;
+        this.y = y;
+        this.character = element;
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: Initalize character in field section

File: 15_2_trail.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    init() {
+        const div = document.createElement('div');
+        div.innerHTML = element;
+        document.body.appendChild(div);
+        array.push(div);
+    }
+
+    update() {
+        array.forEach(item => {
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: I think if u are creating class it should be considered ass standalone part of ecosystem, You should't mutate variables outside of class

File: 15_1_baloon.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+let size = 50;
+let level = -200;
+
+const playGame = () => {
+    clearScore();
+    window.addEventListener('keydown', pump);
+};
+
+const pump = event => {
+    if (event.keyCode === 38) {
```
Comment (grzegorz-bielski) [inline 15_1_baloon.js]: what is `38`? Add it to the `const` and name it appropriately like `upArrowKeyCode` or sth

File: 15_1_baloon.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        maximize();
+    }
+
+    if (event.keyCode === 40) {
+        minimize();
+    }
+
+    if (baloon.style.fontSize === '250px') {
```
Comment (grzegorz-bielski) [inline 15_1_baloon.js:25]: why 250 ?

File: 15_2_trail.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    constructor(x, y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    init() {
+        const random = Math.floor(Math.random() * 25);
+        this.nodeElement = document.createElement('div');
+        this.nodeElement.innerHTML = this.character;
+        this.nodeElement.style.left = this.x + random + 'px';
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: use template string here `${this.x + random}px`

File: 15_1_baloon.js (hunk 7 above)
Comment (magdalena-u) [reply 15_1_baloon.js:25]: there is no particular reason. I've chosen the maximus size.

File: 15_2_trail.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        this.nodeElement.style.top = this.y + random + 'px';
+        this.nodeElement.style.position = 'absolute';
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: that should be added as some css class

File: 15_2_trail.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+const draw = (x, y) => {
+    const star = new Element(x, y);
+    star.init();
+};
+
+class Element {
+    character = '⭐';
+    nodeElement = document.querySelector('div');
```
Comment (AdamusPiotr) [inline 15_2_trail.js]: You can left this not initialized just `nodeElement;`.

File: 15_3_tabs.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+const buttons = document.querySelectorAll('button');
+const arrayTab = [...document.querySelectorAll('tab-panel div')];
+
+const divTabs = arrayTab.map(div => div.getAttribute('data-tabname'));
+
+buttons.forEach(button => {
+    button.addEventListener('click', function() {
+        const currentTab = document.querySelector('.show');
```
Comment (mateusz-bielak) [inline 15_3_tabs.js:7]: Maybe try to segregate expressions inside that event listener using empty lines as we discussed today :)

**Review by mateusz-bielak — APPROVED** [review]
Looks good in my opinion. Please, make adjustments for remaining comments and then you can merge it.

**Pull request merged**